src/meta/src/hummock/compaction/picker/min_overlap_compaction_picker.rs

```diff
@@ -222,127 +221,105 @@ impl NonOverlapSubLevelPicker {
             sstable_infos: vec![vec![]; levels.len()],
         };
         ret.sstable_infos[0].extend(vec![sst.clone()]);
-        let mut overlap_info = self.overlap_strategy.create_overlap_info();
-        let mut select_sst_id_set = BTreeSet::default();
-        #[allow(clippy::single_range_in_vec_init)]
-        let mut overlap_len_and_begins = vec![(sst_index..(sst_index + 1))];
-        for sst in &ret.sstable_infos[0] {
-            overlap_info.update(sst);
-            select_sst_id_set.insert(sst.sst_id);
-        }
 
         for (target_index, target_level) in levels.iter().enumerate().skip(1) {
             if target_level.level_type() != LevelType::Nonoverlapping {
                 break;
             }
 
             // more than 1 sub_level
-            if ret.total_file_count > 1 && ret.total_file_size >= self.max_compaction_bytes
-                || ret.total_file_count >= self.max_file_count as usize
+            if ret.total_file_count > 1
+                && (ret.total_file_size >= self.max_compaction_bytes
+                    || ret.total_file_count >= self.max_file_count as usize)
+            {
+                break;
+            }
```

Comment:
We have the same check in L301. Do we still need this check here?

Reply:
good catch ! Thanks